app/controllers/macros_controller.rb

```diff
@@ -1,8 +1,12 @@
 class MacrosController < ApplicationController
-  before_action :authorize_deployer!
-  before_action :authorize_admin!, only: [:new, :create, :edit, :update, :destroy]
+  include CurrentProject
+  include ProjectLevelAuthorization
 
-  before_action :find_project
+  before_action :authorize_deployer!
+  before_action do
+    find_project(params[:project_id])
+  end
```

Comment:
Does this not have to occur before the above authorize_deployer!? So that it can verify it's a deployer for this project?